File: Mobiles/NewsHub-Android/app/src/main/java/com/newshub/newshub_android/vk/presenter/VkPresenter.kt
package com.newshub.newshub_android.vk.presenter

import com.newshub.newshub_android.App
import com.newshub.newshub_android.general.AppSettings
import com.newshub.newshub_android.vk.VkFragment
import com.newshub.newshub_android.vk.model.VkNews
import com.newshub.newshub_android.vk.model.VkPage
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response


class VkPresenter(val view: VkFragment) {

    fun getPage() {
        App.api?.getVkPage(AppSettings.userId)?.enqueue(object : Callback<VkPage> {
            override fun onResponse(call: Call<VkPage>, response: Response<VkPage>) {
                val page = response.body() ?: return

                view.adapter.setData(page.groups, page.news)
            }

            override fun onFailure(call: Call<VkPage>, t: Throwable) {

            }
        })
    }

    fun getNews(page: Int) {
        App.api?.getVkNews(AppSettings.userId, page)?.enqueue(object : Callback<List<VkNews>> {
            override fun onResponse(call: Call<List<VkNews>>, response: Response<List<VkNews>>) {
                val news = response.body() ?: return

                view.adapter.addNews(news)
            }

            override fun onFailure(call: Call<List<VkNews>>, t: Throwable) {

            }
        })
    }

}

File: Mobiles/NewsHub-iOS/NewsHub-iOS/Vk/Presenter/VkPresenter.swift


class VkPresenter {

    var interactor: VkInteractor?
    var view: VkTableViewController?
    
    
    init(view: VkTableViewController) {
        self.view = view
        self.interactor = VkInteractor()
    }
    
    func getPage() {
        let userId = 2
        interactor?.getPage(userId: userId, { [weak self] (vkPage) in
            var viewData = VkPageView()
            
            for item in vkPage.groups {
                viewData.groups[item.gid] = item
            }
            
            viewData.news = vkPage.news
            self?.view?.setPageData(vkPage: viewData)
        })
    }
    
    func getNews(page: Int) {
        interactor?.getNews(page: page, { [weak self] (vkNews) in
            self?.view?.addNews(news: vkNews)
        })
    }
}
